openhands/core/config/README.md

```diff
@@ -0,0 +1,110 @@
+# Configuration Management in OpenHands
+
+## Overview
+
+OpenHands uses a flexible configuration system that allows settings to be defined through environment variables, TOML files, and command-line arguments. The configuration is managed through a package structure in `openhands/core/config/`.
+
+## Configuration Classes
+
+The main configuration classes are:
+
+- `AppConfig`: The root configuration class
+- `LLMConfig`: Configuration for the Language Model
+- `AgentConfig`: Configuration for the agent
+- `SandboxConfig`: Configuration for the sandbox environment
+- `SecurityConfig`: Configuration for security settings
+
+These classes are defined as dataclasses, allowing for easy type checking and default value specification.
+
+## Loading Configuration from Environment Variables
+
+The `load_from_env` function in the config package is responsible for loading configuration values from environment variables. It recursively processes the configuration classes, mapping environment variable names to class attributes.
+
+### Naming Convention for Environment Variables
+
+- Prefix: Uppercase name of the configuration class followed by an underscore (e.g., `LLM_`, `AGENT_`)
+- Field Names: Uppercase, with underscores replaced by uppercase letters
```

Comment:
What do you mean, "with underscores replaced by uppercase letters"? Underscores exist, they're not replaced. Sonnet confusion? 😆